webapp/components/signup/signup_controller.jsx

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import FormError from 'components/form_error.jsx';
+import LoadingScreen from 'components/loading_screen.jsx';
+
+import UserStore from 'stores/user_store.jsx';
+import BrowserStore from 'stores/browser_store.jsx';
+
+import * as AsyncClient from 'utils/async_client.jsx';
+import Client from 'client/web_client.jsx';
+import * as GlobalActions from 'actions/global_actions.jsx';
+
+import logoImage from 'images/logo.png';
+import ErrorBar from 'components/error_bar.jsx';
+
+import {FormattedMessage} from 'react-intl';
+import {browserHistory, Link} from 'react-router/es6';
+
+export default class SignupController extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.renderSignupControls = this.renderSignupControls.bind(this);
+
+        let loading = true;
+        let serverError = '';
+        let noOpenServerError = false;
+        let usedBefore = false;
+
+        if (window.location.query) {
+            const hash = window.location.query.h;
+            const data = window.location.query.d;
+            const inviteId = window.location.query.id;
+
+            if ((inviteId && inviteId.length > 0) || (hash && hash.length > 0)) {
+                if (UserStore.getCurrentUser()) {
+                    Client.addUserToTeamFromInvite(
```

Comment:
We shouldn't make HTTP requests inside the constructor, can you move this to `componentDidMount`? Also this if statement is a bit confusing, can you try to clean it up a bit?